feat: update document write guide (#48)
bytebase/bytebase.com: docs/document-write-guide.md

---
title: ✍️ Document write guide
order: -1
---

# ✍️ Document write guide

This section attempts to show the easily steps about writing a document. First, let's look at the structure.

## The structure of `.md` file

```markdown
---
title: Document write guide

> order: 40100

Reviewer: This only has 5 digits

---

> # How to write Bytebase document?

Reviewer: ```suggestion
# How to write a document?
```

Write the document content right here.
```

It's a little different from common Markdown file. We need to save its status data into the `--- xxx ---` block. The mainly useful fields are these:

- `title` is the display text in sidebar;

> - `order` indicates the display order. It consists of a array with three two-digit numbers: `{{category}}{{subcategory}}{{document}}`. For example, [Build from Source](/docs/install/build-from-source) is `10200`.

Reviewer: 10200 only has 5 digits

Author: This description has been updated with: `order` is an **integer** representing an array of three two-digit numbers.

  ![corder-example](/docs-assets/order-example.jpg)

## How to write a new document?

1. Write the full content in any editor that supports Markdown. e.g. [VSCode](https://code.visualstudio.com/)/[typora](https://typora.io/)

> 2. Think about where the article should display in sidebar.

Reviewer: ```suggestion
2. Think about where the article should be displayed in sidebar.
```

3. Follow the [example](#the-structure-of-md-file) to figure out the `order` value.
4. Add the meta data block in the top of markdown file.

> 5. Insert the file into the project folder `./docs/`.

Reviewer: ```suggestion
5. Put the file under the project folder `./docs/`.
```

Reviewer: I guess we also need to mention the sub-directory structure determines the hierarchy?
